Atomic/syscontainers.py

```diff
@@ -1199,6 +1165,53 @@ def _do_checkout(self, repo, options):
 
         return options["values"]
 
+    def finalize_checkout(self, rpm_install_content, tmpfiles_template, options, was_service_active):
```

Comment:
looks much better.

Could you prefix the name of these new functions with `_` so it is clear they should be used only within this class?